File: ruby/grains/grains.rb
class Grains
  SQUARES = 64

  def self.square(number)
    unless number.between?(1, SQUARES)
      raise ArgumentError, "Number must be between 1 and #{SQUARES}."
    end
    2**(number - 1)
  end

  def self.total
    (1..SQUARES).sum { |number| square(number) }
  end
end

File: javascript/grains/grains.js
import BigInt from './lib/big-integer';

const NUMBER_OF_SQUARES = 64;

export class Grains {
  square(number) {
    return BigInt(2).pow(number - 1).toString();
  }

  total() {
    let total = BigInt(0);

    for(let i = 1; i <= NUMBER_OF_SQUARES; i++){
      total = total.add(this.square(i));
    }
    return total.toString();
  }
}
